lib/Ix/DBIC/Result.pm

```diff
@@ -102,20 +125,28 @@ sub ix_add_properties ($class, @pairs) {
     Carp::confess("Attempt to add property $name with no data_type")
       unless defined $def->{data_type};
 
-    my $ix_type = $IX_TYPE{ $def->{data_type} };
+    # Is this an array type?
+    if ($def->{data_type} =~ s/^(.*)\[\]$/$1/) {
```

Comment:
nitpick, this code would be easier to skim if:
```perl
if ($def->{data_type} =~ s/\[\]\z//) {
   ...
}
```

No need to preserve `$1` if you just don't touch it.